Internal error in TOR

Caused by: java.lang.StringIndexOutOfBoundsException: String index out of range: 2
at fi.otavanopisto.muikku.plugins.transcriptofrecords.rest.TranscriptofRecordsRESTService.getVops(TranscriptofRecordsRESTService.java:266)

and
at fi.otavanopisto.muikku.plugins.transcriptofrecords.rest.TranscriptofRecordsRESTService.getVops(TranscriptofRecordsRESTService.java:386)

--- a/muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/transcriptofrecords/rest/TranscriptofRecordsRESTService.java
+++ b/muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/transcriptofrecords/rest/TranscriptofRecordsRESTService.java
@@ -262,8 +262,13 @@ public class TranscriptofRecordsRESTService extends PluginRESTService {
                     transferCredit.getGradingScaleIdentifier(),
                     transferCredit.getGradeIdentifier()
                 );
-                if (!StringUtils.isBlank(gradingScaleItem.getName())) {
-                  grade = gradingScaleItem.getName().substring(0, 2);
+                
+                String gradeName = gradingScaleItem.getName();
+                if (!StringUtils.isBlank(gradeName)) {
+                  if (gradeName.length() > 1)
+                    grade = gradeName.substring(0, 2);
+                  else
+                    grade = gradeName;
                 }
               }
               entries.add(new VopsRESTModel.VopsItem(
@@ -380,10 +385,15 @@ public class TranscriptofRecordsRESTService extends PluginRESTService {
                   break;
                 }
                 GradingScaleItem gradingScaleItem = findGradingScaleItemCached(gradingScaleIdentifier, gradeIdentifier);
-                if (!StringUtils.isBlank(gradingScaleItem.getName())) {
+                String gradeName = gradingScaleItem.getName();
+                if (!StringUtils.isBlank(gradeName)) {
                   // 2 characters is enough to cover cases like "10" and "Suoritettu/Saanut opetusta" unambiguously
                   // and still looking good in the matrix
-                  grade = gradingScaleItem.getName().substring(0, 2);
+                  
+                  if (gradeName.length() > 1)
+                    grade = gradeName.substring(0, 2);
+                  else
+                    grade = gradeName;
                 }
                 
                 break;

--- a/muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/transcriptofrecords/rest/TranscriptofRecordsRESTService.java
+++ b/muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/transcriptofrecords/rest/TranscriptofRecordsRESTService.java
@@ -265,7 +265,7 @@ public class TranscriptofRecordsRESTService extends PluginRESTService {
                 
                 String gradeName = gradingScaleItem.getName();
                 if (!StringUtils.isBlank(gradeName)) {
-                  if (gradeName.length() > 1)
+                  if (gradeName.length() > 2)
                     grade = gradeName.substring(0, 2);
                   else
                     grade = gradeName;
@@ -390,7 +390,7 @@ public class TranscriptofRecordsRESTService extends PluginRESTService {
                   // 2 characters is enough to cover cases like "10" and "Suoritettu/Saanut opetusta" unambiguously
                   // and still looking good in the matrix
                   
-                  if (gradeName.length() > 1)
+                  if (gradeName.length() > 2)
                     grade = gradeName.substring(0, 2);
                   else
                     grade = gradeName;

